plugins/modules/monitoring/spectrum_model_attrs.py

```diff
@@ -0,0 +1,526 @@
+        self.success_msg = "Success"
+
+    def build_url(self, path):
+        """
+        Build a sane Spectrum restful API URL
+        :param path: The path to append to the restful base
+        :type path: str
+        :returns: Complete restful API URL
+        :rtype: str
+        """
+
+        return "%s/%s" % (self.url.rstrip('/'), path.lstrip('/'))
+
+    def attr_id(self, name):
+        """
+        Get attribute hex ID
+        :param name: The name of the attribute to retrieve the hex ID for
+        :type name: str
+        :returns: Translated hex ID of name, or None if no translation found
+        :rtype: str or None
+        """
+
+        try:
+            return self.attr_map[name]
+        except KeyError:
+            return None
+
+    def attr_name(self, _id):
+        """
+        Get attribute name from hex ID
+        :param _id: The hex ID to lookup a name for
+        :type _id: str
+        :returns: Translated name of hex ID, or None if no translation found
+        :rtype: str or None
+        """
+
+        for name, m_id in list(self.attr_map.items()):
+            if _id == m_id:
+                return name
+        return None
+
+    def urlencode(self, string):
+        """
+        URL Encode a string
+        :param: string: The string to URL encode
+        :type string: str
+        :returns: URL encode version of supplied string
+        :rtype: str
+        """
+
+        return urlparse.quote(string, "<>%-_.!*'():?#/@&+,;=")
+
+    def update_model(self, model_handle, attrs):
+        """
+        Update a model's attributes
+        :param model_handle: The model's handle ID
+        :type model_handle: str
+        :param attrs: Model's attributes to update. {'<name/id>': '<attr>'}
+        :type attrs: dict
+        :returns: Nothing; exits on error or updates self.results
+        :rtype: None
+        """
+
+        # Build the update URL
+        update_url = self.build_url("/model/%s?" % model_handle)
+        for name, val in list(attrs.items()):
+            if val is None:
+                # None values should be converted to empty strings
+                val = ""
+            val = self.urlencode(str(val))
+            if not update_url.endswith('?'):
+                update_url += "&"
+
+            update_url += "attr=%s&val=%s" % (self.attr_id(name) or name, val)
+
+        # POST to /model to update the attributes, or fail.
+        resp, info = fetch_url(self.module, update_url, method="PUT",
+                               headers={"Content-Type": "application/json",
+                                        "Accept": "application/json"},
+                               use_proxy=self.module.params['use_proxy'])
+        status_code = info["status"]
+        if status_code >= 400:
+            body = info['body']
+        else:
+            body = "" if resp is None else resp.read()
+        if status_code != 200:
+            self.result['msg'] = \
```

Comment:
Same here. (And some more futher below.)

Reply:
fixed.